fix(app release process): enhance add technical user process

manojava-gk opened the pull request:
## Description

enhance user experience by providing table and overlay in add technical user flow. List current users in the table. Allow user to edit the list in an overlay implementation

### Chnagelog
 **Technical Integration - App and Service Release Process**
    - enhance user experience by providing table and overlay in add technical user flow. list current users in the table. allow user to edit the list in an overlay implementation [#1401]

## Why

provide better user experience

## Issue

#1125  #1305 

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/admin/Dev%20Process/How%20to%20contribute.md#commits-branches-and-pull-requests-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

lavanya-bmw commented on src/components/shared/basic/ReleaseProcess/OfferTechnicalIntegration/index.tsx:
```diff
@@ -177,6 +116,36 @@ export default function OfferTechnicalIntegration() {
+
+  const handleUserProfiles = async (roles: string[]) => {
+    console.log(roles)
+    setShowTechUser(false)
```
Please remove console

manojava-gk replied:
removed

lavanya-bmw commented on src/components/shared/basic/ReleaseProcess/TechnicalIntegration/TechUserTable.tsx:
```diff
@@ -0,0 +1,79 @@
+      searchPlaceholder={''}
+      columnHeadersBackgroundColor={'#FFFFFF'}
+      searchDebounce={1000}
+      noRowsMsg={'No tech user added'}
```
Please use translations instead of hardcoded text

manojava-gk replied:
updated

lavanya-bmw reviewed (approved):
lgtm

evegufy commented:
@saadanzari @Cofinity-UX would you also like to review?

Cofinity-UX commented:
Hi @manojava-gk,
here are some points for improving the UX (see screenshot for reference). I'm available to do a quick call, if anything is unclear.

**1. Copywriting for headline:** 
"Create technical user" _(aligns with wording of CTA on main page)_

**2. Copywriting for subline**
"Please provide the requested information to create a new technical user. Your description helps you easily identify different technical users." _(simplified and on point)_

**3. Inputfiel label wording improvement**
"Technical user name"/"Description of technical user"

**4. Placeholder text improvement**
"Enter technical user name"/"Enter technical user description"

**5. Technical user option should not be initially pre-seclected**
_It's not defined which technical user is preferred by the user_

**6. Display 'External technical user' first**
_If it's not possible to display the technical user options after selecting the radio button, 'External technical user' should be displayed as first item, else it will be blocked out of the viewport. Users should see both options initially_

**7. Harmonize font sizes**
_Font size of description of Internal/External technical user should have the same font size (reference size: External technical user)_

**8. Disabled/inactive text should be styled completely as disabled**
_Inactive options in the Internal technical user section should be completely styled as disabled (description of technical users have still active styling)_

**9. Checkbox for External technical user option**
_Identity Wallet Management Technical user should have a checkbox instead of radio button. Since there is only one option: When External technical user is selected, the checkbox for Identity Wallet Management can be preselect


<img width="569" alt="UXreview_technicalusercreation" src="https://github.com/user-attachments/assets/36bd6c32-4157-4516-bce4-06593acaa5fb" />
ed._

manojava-gk commented:
> Create technical user

Hi @Cofinity-UX you are looking at a different screen here. This PR is about technical integration step in app/service release process. The screen that you have shared here is from Technical User page. Will address the points which are valid in the current PR

<img width="939" alt="Screenshot 2025-01-03 at 4 36 51 PM" src="https://github.com/user-attachments/assets/c83e9bc8-b82b-4085-817b-124332afba42" />

<img width="576" alt="Screenshot 2025-01-03 at 4 46 49 PM" src="https://github.com/user-attachments/assets/17b90689-9a1e-40cd-9d14-109b281d5892" />


Here is the detailed response,

1. PFA
2. PFA
3. NA
4. NA
5. Already addressed
6. Updated - PFA
7. Already addressed
8. Updated - PFA
9. In the ticket it is mentioned to create external roles as radio button. [Ticket](https://github.com/eclipse-tractusx/portal-frontend/issues/1305)

Cofinity-UX commented:
@manojava-gk ah, alright, thank you for the clarification!

manojava-gk commented:
@oyo Any review comments for this?
